controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -952,7 +962,12 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 	}
 	// either hardwareData was deleted above, or not found. We need to re-create it
 	if err := r.Client.Create(context.Background(), hd); err != nil {
-		return actionError{errors.Wrap(err, "failed to create hardwareData")}
+		if r.ignoreErrorWhenCleaningModeDisabled(err, info) {
```

Comment:
Same thing here, once a DeletionTimestamp is set there's no reason to ever create this new resource.